src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -0,0 +1,317 @@
+		if (pgSetting == null) {
+			return null;
+		}
+		return pgSetting.getId();
+	}
+	
+	@Override
+	protected Object getUniqueField(SettingConfiguration settingConfiguration) {
+		return settingConfiguration == null ? null : settingConfiguration.getId();
+	}
+	
+	// private Methods
+	private SettingConfiguration convert(org.opensrp.domain.postgres.Settings setting) {
+		if (setting == null || setting.getJson() == null || !(setting.getJson() instanceof SettingConfiguration)) {
+			return null;
+		}
+		return (SettingConfiguration) setting.getJson();
+	}
+	
+	private org.opensrp.domain.postgres.Settings convert(SettingConfiguration entity, Long id) {
+		if (entity == null) {
+			return null;
+		}
+		
+		org.opensrp.domain.postgres.Settings pgSetting = new org.opensrp.domain.postgres.Settings();
+		pgSetting.setId(id);
+		pgSetting.setJson(entity);
+		
+		return pgSetting;
+	}
+	
+	private List<SettingConfiguration> convert(List<org.opensrp.domain.postgres.Settings> settings) {
+		if (settings == null || settings.isEmpty()) {
+			return new ArrayList<>();
+		}
+		
+		List<SettingConfiguration> settingValues = new ArrayList<>();
+		for (org.opensrp.domain.postgres.Settings setting : settings) {
+			SettingConfiguration convertedSetting = convert(setting);
+			if (convertedSetting != null) {
+				settingValues.add(convertedSetting);
+			}
+		}
+		return settingValues;
+	}
+	
+	private SettingsMetadata createMetadata(SettingConfiguration entity, Long id) {
+		SettingsMetadata metadata = new SettingsMetadata();
+		metadata.setSettingsId(id);
+		metadata.setDocumentId(entity.getId());
+		metadata.setIdentifier(entity.getIdentifier());
+		metadata.setServerVersion(entity.getServerVersion());
+		return metadata;
+	}
+	
+	@Override
+	public SettingsMetadata saveSetting(SettingConfiguration entity, SettingsMetadata metadata) {
+		if (entity == null || entity.getIdentifier() == null) {
+			return null;
+		}
+		
+		if (entity.getId() == null) {
+			entity.setId(UUID.randomUUID().toString());
+		}
+		setRevision(entity);
+		
+		Settings settings = convert(entity, Long.valueOf(entity.getId()));
+		if (settings == null) {
+			return null;
+		}
+		
+		int rowsAffected = entity.getId() != null ? settingMapper.updateByPrimaryKeySelective(settings)
+		        : settingMapper.insertSelectiveAndSetId(settings);
+		if (rowsAffected < 1 || settings.getId() == null) {
+			return null;
+		}
+		
+		SettingsMetadata settingsMetadata = metadata != null ? metadata : createMetadata(entity, settings.getId());
+		settingsMetadata.setServerVersion(Calendar.getInstance().getTimeInMillis());
+		if (settingsMetadata != null) {
+			
+			if (settingsMetadata.getId() != null) {
+				
+				settingMetadataMapper.updateByPrimaryKeySelective(settingsMetadata);
+			} else {
+				
+				settingMetadataMapper.insertSelective(settingsMetadata);
+			}
+		}
+		
+		return settingsMetadata;
+		
+	}
+	
+	@Override
+	public void add(SettingConfiguration entity) {
```

Comment:
This method seems to be a duplicate of  `addSetting(SettingConfiguration entity)` in this class.

One of them can be removed.